Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Bugzilla41424.cs

```diff
@@ -67,19 +68,37 @@ protected override void Init()
 		public void DatePickerCancelShouldUnfocus()
 		{
 			RunningApp.Tap(q => q.Marked(DatePicker));
-			RunningApp.Back();
+			Assert.IsTrue(DialogIsOpened(),"Tap Picker");
 
 			RunningApp.WaitForElement(q => q.Marked("Click to view focus state"));
 			RunningApp.Tap(q => q.Marked("Click to view focus state"));
 			RunningApp.WaitForElement(q => q.Marked("unfocused"));
 
 			RunningApp.Tap(q => q.Marked("Click to focus DatePicker"));
-			RunningApp.Back();
+			RunningApp.WaitForElement(q => q.Marked("OK"));
+			Assert.IsTrue(DialogIsOpened(),"Call Focus Picker");
 
 			RunningApp.WaitForElement(q => q.Marked("Click to view focus state"));
 			RunningApp.Tap(q => q.Marked("Click to view focus state"));
 			RunningApp.WaitForElement(q => q.Marked("unfocused"));
 		}
+
+		bool DialogIsOpened()
```

Comment:
A couple of requests:
1. Can we make this an extension method that takes `this App` (i.e., RunningApp.DialogIsOpened()`) so we can use it in other places? It seems like a generally useful method.
2. Can we add a timeout parameter (int timeout = 1500)?
3. Can we have it poll every few hundred milliseconds until the timeout? That way it _could_end ealier than 1.5 seconds, if the dialog is already open.

Not required for getting this merged, just a nice-to-have.